src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5455,6 +5467,12 @@ private void ParseTypeArgumentList(out SyntaxToken open, SeparatedSyntaxListBuil
             }
 
             close = this.EatToken(SyntaxKind.GreaterThanToken);
+
+            void eatCommaAndTypeArgument()
+            {
+                types.AddSeparator(this.EatToken(SyntaxKind.CommaToken));
+                types.Add(this.ParseTypeArgument());
+            }
```

Comment:
i don't see a lot of value with this helper.  i think it would be easier to understand if it was just inlined.